addons/l10n_latam_invoice_document/models/account_move.py

```diff
@@ -212,7 +212,11 @@ def _compute_l10n_latam_document_type(self):
                 document_types = document_types.filtered(lambda x: x.internal_type not in ['credit_note'])
             if rec.debit_origin_id:
                 document_types = document_types.filtered(lambda x: x.internal_type == 'debit_note')
-            rec.l10n_latam_document_type_id = document_types and document_types[0].id
+
+            if not document_types:
+                rec.l10n_latam_document_type_id = False
+            elif rec.l10n_latam_document_type_id not in document_types:
+                rec.l10n_latam_document_type_id = document_types[0].id
```

Comment:
```suggestion
          if rec.l10n_latam_document_type_id not in document_types:
               rec.l10n_latam_document_type_id = document_types and document_types[0].id
```

Is not that the same?  Otherwise, the idea seems fine, although more improvement than fix. 
```

>>> self.env['res.users'] in self
False

```